.github/workflows/code-validation.yaml

```diff
@@ -33,11 +33,23 @@ jobs:
         python-version: '3.x'
         cache: pip
 
+    - name: Download lychee binary (for precommit test)
+      if: matrix.test == 'precommit'
```

Comment:
This one took less than one second to run, but doesn't have any cache setup so we could arguably leave it as is, to prevent some download flakiness.

```suggestion
```